### ISSUE TITLE
tzwin.py fails to import

### ISSUE THREAD

**Issue by labrys** [dateutil/dateutil#133]
`dateutil/tz.py` never uses `tzwin.py` since `tzwin.py` fails to import with the following exception:
```
Traceback (most recent call last):
  File "dateutil/tzwin.py", line 7, in <module>
    from .tz import tzname_in_python2
ValueError: Attempted relative import in non-package
```

**Comment by pganssle** [comment]
@labrys Can you provide an example that shows how this came up? What OS are you using? What version of python are you using?

**Comment by labrys** [comment]
This came up during troubleshooting of an issue with SickRage's date handling while trying to display a TV show's air times by the timezone of a network.  SickRage is running with Python 2.7.10 but I have confirmed this with a simple test on Python 2.7.10 and Python 3.4.3.

When the TV show's network is None, the SickRage code works on Debian but not Windows (tested on Win 10 x64, Win 8.1 x64, Win XP x86 SP3).  In SickRage, `network_timezones.py` sets a variable 'sb_timezone=tz.tzlocal()' as a fallback when the network's timezone was `None` or could not be parsed.  This seemed to be the failing point on Windows Systems.

So, while looking at 'dateutil' to see how Windows differed, I found that in 'tz.py' during the import ?(shown below) `tzwin` was never being imported due to an ImportError in `tzwin.py` that has since been fixed in dateutil (with `from six.moves import winreg').  
```python
try:
    from dateutil.tzwin import tzwin, tzwinlocal
except (ImportError, OSError):
    tzwin, tzwinlocal = None, None
``` 
SickRage updated `dateutil` from version 2.2 to 2.4.2. but the error persisted.  So I looked at the changes to `dateutil` and found that `tzwin.py` still wasn't importing which can be verified with
```python
import dateutil.tz as tz
print(tz.tzwinlocal)
```
Which should output `<class 'dateutil.tzwin.tzwinlocal'>` but instead returns `None`
The cause is a failure import `tzname_in_python2() ` due to the above mentioned exception.  However on normal usage of dateutil you don't import `tzwin.py` directly and the exception gets swallowed during the `tz.py` try:except clause during import of `tzwin`

ref: (SiCKRAGETV/sickrage-issues/issues/3007)

**Comment by pganssle** [comment]
OK, I'll look at this in a bit and try and get it sorted out.

**Comment by labrys** [comment]
I issued a pull request for a proposed fix.
Pulled `tzname_in_python2` out of `tz.py` and into a new `tzcommon.py` and added imports to each

**Comment by pganssle** [comment]
I pulled this up in Python 2.7.10 and Python 3.4 on Windows and I could not reproduce it. This may be a problem with your specific installation.

**Comment by labrys** [comment]
So, just to confirm you can sucessfully run the following in Python 2.7.10:

`c:\python27\python -c "import dateutil.tz; print dateutil.tz.tzwinlocal()"`

Because for me, on 3 separate Windows systems with a fresh clone of dateutil it returns:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: 'NoneType' object is not callable
```

**Comment by labrys** [comment]
Oh and just for clarification the import error initially mentioned gets suppressed when importing `dateutil.tz` during normal usage due to the 'try: ... except ImportError` clause in `dateutil.tz`

**Comment by pganssle** [comment]
Correct. In both Python 2.7.10 and Python 3.4, I get:

```
python -c "import dateutil.tz as tz; print(tz.tzwinlocal)"
<class 'dateutil.tzwin.tzwinlocal>
```

What do you mean by "a fresh clone of dateutil"? Are you running directly from the `dateutil` directory, or are you installing it by running `python setup.py install`?

I even tried downloading the 2.4.2 release tarball and running directly from the directory containing `setup.py` and got the same result, in both Python 3.4 and Python 2.7.10. I think this is some configuration problem on your end.

**Comment by pganssle** [comment]
@labrys Can you check if you can run [this test](https://github.com/dateutil/dateutil/blob/master/dateutil/test/test.py#L5963) on your Windows system?

**Comment by labrys** [comment]
Tested good in Win 10 x64 for both Python 2.7.10 and Python 3.4

**Comment by labrys** [comment]
And to answer your question the initial testing was done using the copies of dateutil package in SickRage, however since I have also done a `pip install python-dateutil` and run the tests and I have used git clone to fork a dev copy then run setup.py and run from the dateutil directory and all methods fail identically.

**Comment by labrys** [comment]
Oh and this is both using PyCharm 4.5.4 or running from command prompt.

**Comment by labrys** [comment]
I'll make a clean Windows VM later today if I get a chance and test again.  What method of installation do you recommend.  I assume `pip install python-dateutil`.

**Comment by pganssle** [comment]
That's how I'd usually do it, but on my side it's working even if I just download the tarball, so I don't know exactly why it's broken on your end.

I don't understand why that test works for you - it relies on importing `tzwin`, so it should be throwing an error if you can't import tzwin.

**Comment by labrys** [comment]
The test works because I can import `tzwin` without issues.  It's `tz` that can't import `tzwin` because of the recursive reference so it defaults setting `tzwin = tzwinlocal = None`

**Comment by pganssle** [comment]
Oh, I found the problem. This change was made *after* 2.4.2, so pulling down the 2.4.2 release doesn't have the problem. Confirmed this on my end.

This requires a bit of refactoring, but that shouldn't be a problem. I'll accept your other PR, but I think maybe the right way to do this to refactor the `tz` stuff into a submodule and deprecate direct access to `dateutil.tzwin`. I'll do the refactoring based on your PR, though.

**Comment by labrys** [comment]
Sounds great

**Comment by pganssle** [comment]
Fix merged in #141

**Comment by labrys** [comment]
Confirmed working on Windows with various python releases.
